src/Analyzer.Cli/CommandLineParser.cs

```diff
@@ -41,39 +43,70 @@ private RootCommand SetupCommandLineAPI()
 
-            Option<FileInfo> templateOption = new Option<FileInfo>(
-                    "--template-file-path",
-                    "The ARM template to analyze")
-            {
-                IsRequired = true
-            };
-            templateOption.AddAlias("-t");
-            rootCommand.AddOption(templateOption);
+            // Setup analyze-template w/ template file argument and parameter file option
+            Command analyzeTemplateCommand = new Command(
+                "analyze-template",
+                "Analyze a singe template");
+            
+            Argument<FileInfo> templateArgument = new Argument<FileInfo>(
+                "template-file-path",
+                "The ARM template to analyze");
+            analyzeTemplateCommand.AddArgument(templateArgument);
 
             Option<FileInfo> parameterOption = new Option<FileInfo>(
                  "--parameters-file-path",
                  "The parameter file to use when parsing the specified ARM template");
             parameterOption.AddAlias("-p");
-            rootCommand.AddOption(parameterOption);
+            analyzeTemplateCommand.AddOption(parameterOption);            
+            
+            analyzeTemplateCommand.Handler = CommandHandler.Create<FileInfo, FileInfo>((templateFilePath, parametersFilePath) => this.AnalyzeTemplate(templateFilePath, parametersFilePath));
+
+            // Setup analyze-directory w/ directory argument 
+            Command analyzeDirectoryCommand = new Command(
+                "analyze-directory", 
+                "Analyze all templates within a directory");
+
+            Argument<DirectoryInfo> directoryArgument = new Argument<DirectoryInfo>(
+                "directory-path",
+                "The directory to analyze");
+            analyzeDirectoryCommand.AddArgument(directoryArgument);
 
-            rootCommand.Handler = CommandHandler.Create<FileInfo, FileInfo>((templateFilePath, parametersFilePath) => this.RootCommandHandler(templateFilePath, parametersFilePath));
+            analyzeDirectoryCommand.Handler = CommandHandler.Create<DirectoryInfo>((directoryPath) => this.AnalyzeDirectory(directoryPath));
+
+            // Add commands to root command
+            rootCommand.AddCommand(analyzeTemplateCommand);
+            rootCommand.AddCommand(analyzeDirectoryCommand);
 
             return rootCommand;
         }
 
-        private void RootCommandHandler(FileInfo templateFilePath, FileInfo parametersFilePath)
+        private void AnalyzeTemplate(FileInfo templateFilePath, FileInfo parametersFilePath, bool analyzeSingleTemplate = true)
         {
             try
             {
-                var templateAnalyzer = new Core.TemplateAnalyzer(File.ReadAllText(templateFilePath.FullName), parametersFilePath == null ? null : File.ReadAllText(parametersFilePath.FullName), templateFilePath.FullName);
+
+                string templateFileContents = File.ReadAllText(templateFilePath.FullName);
```

Comment:
Can you add some validation for when a single file is passed and it's not a json file.